Add docs for elastic-agent diagnostics collect (#1176)
elastic/observability-docs: docs/en/ingest-management/commands.asciidoc

Gather diagnostics information from the {agent} and applications it's running.

> If no options are specified, this command will display version numbers and application meta-data.

Reviewer: ```suggestion
If no options are specified, this command displays version numbers and application metadata.
```

> If the `collect` option is specified, it will produce an archive with this meta-data, configuration information, the policy, as well as any local logs.

Reviewer: ```suggestion
If `collect` is specified, it produces an archive containing application metadata, configuration information, the policy, and any local logs.
```

> Note that the archive *will not redact credentials*, they may appear in plain text in the configuration or policy files inside the archive.

Reviewer: ```suggestion
Note that *credentials are not redacted* in the archive; they may appear in plain text in the configuration or policy files inside the archive.
```

[discrete]
=== Synopsis

[source,shell]
----
elastic-agent diagnostics [--help] [--output <string>] [global-flags]
elastic-agent diagnostics collect [--output <string>]
                                  [--file <string>]
                                  [--help]
                                  [global-flags]
----

[discrete]
=== Options

`--output <string>`::

> Output format. Specify `human`, `json`, or `yaml`. The default is `human`.
> If using the `collect` option, specify `yaml`, or `json`. The default in this case is `yaml`.
> 

Reviewer: ```suggestion
Output format. If using `collect`, specify `json` or `yaml` (the default).
Otherwise specify `json`, `yaml`, or `human` (the default).

```

Reviewer: Swapped things around because I think it's better to put something that has a condition first in the list (for folks who skim read).

For the other bits, I think it makes more sense to put the default setting at the end because users are less likely to explicitly set the value if it's a default. I don't love the parens, but maybe worth using here to avoid repeating "the default is" twice in the description. WDTY?

`--file`::
Output archive name for the `collect` option.

> Defaults to "elastic-agent-diagnostics-<timestamp>.zip" where the timestamp is the current time in UTC.

Reviewer: ```suggestion
Defaults to `elastic-agent-diagnostics-<timestamp>.zip` where the timestamp is the current time in UTC.
```
